libfoundation/src/foundation-data.cpp

```diff
@@ -779,11 +779,16 @@ bool MCDataFirstIndexOf(MCDataRef p_data, MCDataRef p_chunk, MCRange t_range, ui
 {
 	__MCAssertIsData(p_data);
 	__MCAssertIsData(p_chunk);
+	
+	if (MCDataGetLength(p_data) == 0)
+		return false;
```

Comment:
I suggest removing this check and replacing it with a t_range.length == 0 check after the __MCDataClampRange call.